src/main/java/duke/main/Parser.java

```diff
@@ -0,0 +1,103 @@
+package duke.main;
+
+import duke.command.*;
+import duke.exception.DukeException;
+
+public class Parser {
+    public Parser() {
+    }
+
+    public Command parse(String[] fullCommand) throws DukeException {
```

Comment:
Can consider switching to a static method as no instance of Parser is required to run this method. If so, we can modify the way we access `parse()` in Duke.java by accessing it through the class `Parser` rather than an instance `parser`.